python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/utils/asset_condition_scenario.py

```diff
@@ -0,0 +1,90 @@
+import dataclasses
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+import dagster._check as check
+from dagster import AssetKey
+from dagster._core.definitions.asset_condition.asset_condition import (
+    AssetCondition,
+    AssetConditionEvaluationState,
+    AssetConditionResult,
+)
+from dagster._core.definitions.asset_condition.asset_condition_evaluation_context import (
+    AssetConditionEvaluationContext,
+)
+from dagster._core.definitions.asset_daemon_context import AssetDaemonContext
+from dagster._core.definitions.events import CoercibleToAssetKey
+from dagster._core.instance import DagsterInstance
+from dagster._seven.compat.pendulum import pendulum_freeze_time
+from dagster._utils.caching_instance_queryer import CachingInstanceQueryer
+
+from .asset_scenario_state import AssetScenarioState
+
+
+@dataclass(frozen=True)
+class AssetConditionScenarioState(AssetScenarioState):
+    asset_condition: Optional[AssetCondition] = None
+    previous_evaluation_state: Optional[AssetConditionEvaluationState] = None
+
+    @staticmethod
+    def create_from_state(
+        state: AssetScenarioState, asset_condition: AssetCondition
+    ) -> "AssetConditionScenarioState":
+        return AssetConditionScenarioState(
+            **{
+                **{f.name: getattr(state, f.name) for f in dataclasses.fields(state)},
```

Comment:
Not a huge fan of this cleverness (would rather just list these things out which is obvious and you get typechecking)

Also does it ever make sense to create an  `AssetConditionScenarioState` from the daemon variant?

Reply:
It would not really make sense to create an AssetConditionScenarioState from the daemon variant -- this is basically here so we can have a "library" of common setups (dependency relationships etc.) that either subclass can pull from. This is more evidence of it being nicer to go with an approach in which AssetConditionScenarioState contains an AssetScenarioState rather than it being a subclass.

That also gets around this weird field copying approach (which I went with out of mild fear of adding a new field to the parent class and forgetting to add it to this constructor).